nodeup/pkg/model/kubelet.go

```diff
@@ -705,7 +705,7 @@ func (b *KubeletBuilder) buildKubeletConfigSpec(ctx context.Context) (*kops.Kube
 	// In certain configurations systemd-resolved will put the loopback address 127.0.0.53 as a nameserver into /etc/resolv.conf
 	// https://github.com/coredns/coredns/blob/master/plugin/loop/README.md#troubleshooting-loops-in-kubernetes-clusters
 	if c.ResolverConfig == nil {
-		if b.Distribution.HasLoopbackEtcResolvConf() {
+		if b.Distribution.HasLoopbackEtcResolvConf(c.CloudProvider) {
```

Comment:
We had a chat about this during our last office hours and we wanted to try just checking if `/run/systemd/resolve/resolv.conf` exists. If it does exist, use it. What do you think about this approach?

Reply:
Hi, sounds like a good approach. Let's give it a try